test/lib/general.lib.test.js

```diff
@@ -48,4 +48,49 @@ describe('RequestLib', () => {
       expect(result).to.equal('2015-10-21T20:31:57.000Z')
     })
   })
+
+  describe('#periodsOverlap', () => {
```

Comment:
I think to cover all bases there are a few more scenarios we could add that would hopefully answer any questions someone had about what this function does in certain scenarios.

These are my suggestions for the scenarios to cover

- when there is no overlap (you've already got this)
  - `referencePeriod = [{ startDate: new Date('2023-02-01'), endDate: new Date('2023-02-28') }]`
  - `checkPeriod = [{ startDate: new Date('2023-01-01'), endDate: new Date('2023-01-31') }]`
- when a check period overlaps the start of a reference period
  - `referencePeriod = [{ startDate: new Date('2023-02-01'), endDate: new Date('2023-02-28') }]`
  - `checkPeriod = [{ startDate: new Date('2023-01-15'), endDate: new Date('2023-02-15') }]`
- when a check period overlaps the end of a reference period
  - `referencePeriod = [{ startDate: new Date('2023-01-01'), endDate: new Date('2023-01-31') }]`
  - `checkPeriod = [{ startDate: new Date('2023-01-15'), endDate: new Date('2023-02-15') }]`
- when a reference period is _completely_ inside a check period
  - `referencePeriod = [{ startDate: new Date('2023-02-01'), endDate: new Date('2023-02-15') }]`
  - `checkPeriod = [{ startDate: new Date('2023-01-01'), endDate: new Date('2023-02-28') }]`
- when a check period is _completely_ inside a reference period
  - `referencePeriod = [{ startDate: new Date('2023-01-01'), endDate: new Date('2023-02-28') }]`
  - `checkPeriod = [{ startDate: new Date('2023-02-01'), endDate: new Date('2023-02-15') }]`
- when the periods are the same
  - `referencePeriod = [{ startDate: new Date('2023-02-01'), endDate: new Date('2023-02-28') }]`
  - `checkPeriod = [{ startDate: new Date('2023-02-01'), endDate: new Date('2023-02-28') }]`